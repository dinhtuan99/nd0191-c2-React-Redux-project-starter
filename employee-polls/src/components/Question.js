import { connect } from "react-redux";
import moment from 'moment'
import { Link } from "react-router-dom";


const Question = ({ question }) => {

  const time = new Date(question.timestamp)
  const timeStr = moment(time).format('hh:mm:A | MM/DD/YYYY')


  return (
    <div className="col d-flex justify-content-center">
      <div className="card" style={{ width: "18rem" }}>
        <div className="card-body">
          <h5 className="card-title">{ question.author }</h5>
          <h6 className="card-subtitle mb-2 text-body-secondary">
          { timeStr }
          </h6>
          <Link to={`/questions/${question.id}`} className="btn btn-primary w-100">Show</Link>
        </div>
      </div>
    </div>
  );
};

export default connect()(Question);
