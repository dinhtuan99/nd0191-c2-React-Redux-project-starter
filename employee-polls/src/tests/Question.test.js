import * as React from "react";
import { render } from "@testing-library/react";
import Question from "../components/Question";
import middleware from "../middleware";
import reducer from "../reducers";
import { createStore } from "redux";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";

const store = createStore(reducer, middleware);

describe("Question", () => {
  it("matches the snapshot when a name is passed", () => {
    const question = {
      id: "xj352vofupe1dqz9emx13r",
      author: "mtsamis",
      timestamp: 1493579767190,
      optionOne: {
        votes: ["mtsamis", "zoshikanlu"],
        text: "deploy to production once every two weeks",
      },
      optionTwo: {
        votes: ["tylermcginnis"],
        text: "deploy to production once every month",
      },
    };
    var view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Question question={question} />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
