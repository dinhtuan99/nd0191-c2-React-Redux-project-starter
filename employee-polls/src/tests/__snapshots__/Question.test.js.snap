// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question matches the snapshot when a name is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col d-flex justify-content-center"
      >
        <div
          class="card"
          style="width: 18rem;"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              mtsamis
            </h5>
            <h6
              class="card-subtitle mb-2 text-body-secondary"
            >
              02:16:AM | 05/01/2017
            </h6>
            <a
              class="btn btn-primary w-100"
              href="/questions/xj352vofupe1dqz9emx13r"
            >
              Show
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col d-flex justify-content-center"
    >
      <div
        class="card"
        style="width: 18rem;"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            mtsamis
          </h5>
          <h6
            class="card-subtitle mb-2 text-body-secondary"
          >
            02:16:AM | 05/01/2017
          </h6>
          <a
            class="btn btn-primary w-100"
            href="/questions/xj352vofupe1dqz9emx13r"
          >
            Show
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
